src/libraries/System.Console/src/System/ConsolePal.Windows.cs

```diff
@@ -1186,7 +1186,7 @@ private static unsafe int ReadFileNative(IntPtr hFile, Span<byte> buffer, bool i
 
             private static unsafe int WriteFileNative(IntPtr hFile, ReadOnlySpan<byte> bytes, bool useFileAPIs)
             {
-                // You can't use the fixed statement on an array of length 0.
+                // The fixed statement maps an empty ReadOnlySpan to nullptr.
```

Comment:
This comment does not make sense. Both WriteFile and WriteConsole APIs below handle `lpBuffer: NULL` with `nNumberOfBytesToWrite: 0` just fine. It would be better to delete this comment.

As it stands, this is a performance optimization that avoids the syscall when there is nothing to do.

Reply:
Same for the other ones.